src/chain/chain.rs

```diff
@@ -495,22 +495,19 @@ impl Chain {
     }
 
     /// Return the keys we know
-    pub fn get_their_keys_info(&self) -> impl Iterator<Item = (&Prefix<XorName>, &TheirKeyInfo)> {
+    pub fn get_their_keys_info(&self) -> impl Iterator<Item = (&Prefix<XorName>, &SectionKeyInfo)> {
         self.state.get_their_keys_info()
     }
 
     /// Returns `true` if the `proof_chain` contains a key we have in `their_keys` and that key is
     /// for a prefix compatible with `prefix`
     pub fn check_trust(&self, prefix: &Prefix<XorName>, proof_chain: &SectionProofChain) -> bool {
-        let filtered_keys: BTreeSet<_> = if prefix.is_compatible(self.our_prefix()) {
-            self.state.our_history.all_keys().collect()
-        } else {
-            self.state
-                .get_their_keys_info()
-                .filter(|&(pfx, _)| prefix.is_compatible(pfx))
-                .map(|(_, info)| &info.key)
-                .collect()
-        };
+        let filtered_keys: BTreeSet<_> = self
+            .state
+            .get_their_keys_info()
+            .filter(|&(pfx, _)| prefix.is_compatible(pfx))
```

Comment:
just to confirm, so this means when checking trust, we always use their_keys from the shared_state.
It seems this is the main place that `our_history.all_keys()` to be used.
If so, do we still need it?

Reply:
proof_chain just bellow here is the same type and we use `all_keys()`